server/lib/nl/nl_utils.py

```diff
@@ -89,4 +92,107 @@ def combine_stop_words() -> Set[str]:
 def remove_punctuations(s):
   s = s.replace('\'s', '')
   s = re.sub(r'[^\w\s]', ' ', s)
-  return " ".join(s.split())
\ No newline at end of file
+  return " ".join(s.split())
+
+
+def is_topic(sv):
+  return sv.startswith("dc/topic/")
+
+
+def is_svg(sv):
+  return sv.startswith("dc/g/")
+
+
+def is_sv(sv):
+  return not (is_topic(sv) or is_svg(sv))
+
+
+#
+# Returns a list of existing SVs (as a union across places).
+#
+def sv_existence_for_places(places: List[str], svs: List[str]) -> List[str]:
+  if not svs:
+    return []
+
+  sv_existence = dc.observation_existence(svs, places)
+  if not sv_existence:
+    logging.error("Existence checks for SVs failed.")
+    return []
+
+  existing_svs = set()
+  for sv in svs:
+    for _, exist in sv_existence['variable'][sv]['entity'].items():
+      if not exist:
+        continue
+      existing_svs.add(sv)
+
+  return list(existing_svs)
+
+
+_NUM_CHILD_PLACES_FOR_EXISTENCE = 20
```

Comment:
super nit: all constants at the top of the file?

Reply:
Done.